tools/pv_tree_generator/build_tree.py

```diff
@@ -1,243 +1,329 @@
+            'populationType': pop_type,
+            'sv': [],
+            'l': text_format.format_title(self.value),
+            't': 'v',
+            'e': self.value,
+            'c': 0,
+            'sv_set': set([]),
+            'cd': [],
+        }
+        for node in self.leafs:
+            if node.addLevel:
                 value_blob = {
-                    'populationType': value_ui_node.pop_type,
-                    'sv': [sv.dcid],
-                    'l': text_format.format_title(value_ui_node.text),
+                    'populationType': node.pop_type,
+                    'sv': node.dcids,
+                    'l': text_format.format_title(node.name),
                     't': 'v',
-                    'e': value_ui_node.enum,
+                    'e': node.name,
                     'c': 1,
-                    'sv_set': set([sv.dcid]),
+                    'cd': [],
+                    'sv_set': set(node.dcids)
                 }
-                if sv.se:
-                    value_blob['se'] = sv.se
-                # add statistical variables as the child of current node
                 result['cd'].append(value_blob)
-
-                if level <= MAX_LEVEL:
-                    # build the branches recursively
-                    for child in child_pos:
-                        branch = build_tree_recursive(
-                            child, level + 1, pop_obs_spec, stat_vars, value_ui_node)
-                        if branch['cd']:
-                            if 'cd' not in value_blob:
-                                value_blob['cd'] = []
-                            value_blob['cd'].append(branch)
-                        value_blob['sv_set'] |= branch['sv_set']
-                        del branch['sv_set']
-                value_blob['c'] = len(value_blob['sv_set'])
             else:
-                # if this is a duplicated statsVar
-                # append the statsVar to the right node, increase the count by 1
-                for child in result['cd']:
-                    if child['e'] == sv.pv[property_diff]:
-                        child['sv'].append(sv.dcid)
+                result['sv'].extend(node.dcids)
+            result['sv_set'].update(node.dcids)
+        if not result['sv']:
+            result['t'] = 'p'
+        return result
 
-    result['cd'] = text_format.filter_and_sort(property_diff,
-                                               result['cd'], False)
 
-    # update the count
-    if result['cd']:
-        for child in result['cd']:
-            result['sv_set'] |= child['sv_set']
-            del child['sv_set']
+class valueLeaf:
+    def __init__(self, name, dcids, addLevel, pos):
+        self.name = name
+        self.dcids = dcids
+        self.addLevel = addLevel
+        self.pop_type = pos.pop_type
 
-    result['c'] = len(result['sv_set'])
-    result = group_super_enum(result)
-    return result
 
+def build_tree(max_level):
+    global POP_OBS_SPEC
```

Comment:
global is not recommended. Can put them in a separate module if you want to share it across other modules.

Reply:
Removed global. Pass it to class PropertyNode and VariableNode as arguments, removed the argument 'vertical' in PropertyNode with this change.